### PR TITLE
feat: add vault non-root support

### PR DESCRIPTION
**Pull request by urfreespace**
https://github.com/streamnative/charts/issues/840

Add vault non-root user vault (100:1000) support

```
vault:
  replicaCount: 1
  securityContext:  
    runAsUser: 100
    runAsGroup: 1000
    supplementalGroups: [100]
    fsGroup: 1000

images:
  vault_init:
    repository: urfreespace/pulsar_vault_init
    tag: "non-root-4"
```

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/templates/vault/vault-initialize-public-key.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ spec:
            echo "vault root token is ready";
            rootToken=$(kubectl get secret {{ template "pulsar.vault-unseal-secret-key-name" . }} -o jsonpath='{.data.vault-root}'|base64 -d);
            echo $rootToken;
-           echo $rootToken > /root/rootToken;
+           echo $rootToken > /vault/rootToken;
```
Comment (fantapsody) [inline charts/sn-platform/templates/vault/vault-initialize-public-key.yaml:53]: How will `/vault/rootToken` be used?

**Review by MarvinCai — APPROVED** [review]
if we need to build custom image for vault,

File: charts/sn-platform/templates/vault/vault-initialize-public-key.yaml (hunk 1 above)
Comment (urfreespace) [reply charts/sn-platform/templates/vault/vault-initialize-public-key.yaml:53]: non-root users can't access /root directory, and from vault Dockerfile https://github.com/hashicorp/docker-vault/blob/cba1f2209c3bddb1319c56b17ef308a2c6cb6c7c/ubi/Dockerfile#L66 we know vault has a home directory for user vault

**Comment by urfreespace** [comment]
> 

Our vault-init image is based on the office vault image, and just added `kubectl` in, for now, it can meet the requirement

**Review by ericsyh — APPROVED** [review]

**Comment by maxsxu** [comment]
> > 
> 
> Our vault-init image is based on the office vault image, and just added `kubectl` in, for now, it can meet the requirement

What does the kubectl inside this vault image used for? @urfreespace

**Comment by urfreespace** [comment]
> > > 
> > 
> > 
> > Our vault-init image is based on the office vault image, and just added `kubectl` in, for now, it can meet the requirement
> 
> What does the kubectl inside this vault image used for? @urfreespace

manage the relationship between vault super-token and the secret, I think, maybe @tuteng could know more

**Comment by tuteng** [comment]
@fantapsody PTAL

File: charts/sn-platform/templates/vault/vault-initialize-public-key.yaml (hunk 1 above)
Comment (ericsyh) [reply charts/sn-platform/templates/vault/vault-initialize-public-key.yaml:53]: @fantapsody PTAL

**Review by fantapsody — APPROVED** [review]

**Pull request merged**